### ISSUE TITLE
When trying to install getting error package missing python

### ISSUE THREAD

**Issue by adispi** [SynoCommunity/spksrc#3310]
when trying to install packages that relay on python i get an error that the python package is missing.
i have tried to install the synology package and the package for synocommunity.

thank you.
adi

**Comment by davidmahon2** [comment]
I have the same problem updating SABnzbd to the latest synocommunity release on a DS918+.  Python 2.7.14-17 is already installed.

**Comment by sibyy** [comment]
Same issue with DS918+. I have Python 2.7.14.18.

**Comment by ymartin59** [comment]
Please precise what package is concerned and where you got it from, either online repository or from #3138

**Comment by davidmahon2** [comment]
In my case, the packages were installed from [https://github.com/SynoCommunity/spksrc/issues/3138](url)

**Comment by sibyy** [comment]
I got it from 3138.
It's actually: python_x64-6.1_2.7.14-18.spk

Thanks

**Comment by adispi** [comment]
i am using the synocommunity repository.
i have tried sabnbzd 2.3.3-34
and mylar 20160119-4
on DS918+

**Comment by justchil** [comment]
I was able to install manually and everything appears to be working fine on my DS918+.

Installed:
python_x64-6.1_2.7.14-18.spk
sabnzbd_x64-6.1_2.3.3-34.spk

From:
https://github.com/Safihre/spksrc/releases/

**Comment by RoketteS** [comment]
Does anybody has any luck with?
python_x64-6.1_2.7.14-18.spk
couchpotatoserver_noarch-all_20180228-7.spk

Manually I'm still getting the same error

**Comment by raspberrycoulis** [comment]
Since upgrading my NAS from the DS215j to the DS718+, I haven’t been able to get CP running either. Complains about Python missing even though it is installed. I also manually installed the package @RoketteS mentioned.

**Comment by dallisjcjc** [comment]
I am having the same problem.  Brand new DS918+ successfully installed and running Python3 v3.5.2-6  which I installed from synoCommunity packages, if I try to install:
-CouchPotato v20180513-7 
-SABnzbd v2.3.3-34 
-SickBeard Custom v20151220-4

Also tried installing above packages unsuccessfully after successfully uninstalling Python 3 above and installing and running python_x64-6.1_2.7.14-18.spk from https://github.com/Safihre/spksrc/releases/

**Comment by raspberrycoulis** [comment]
Yes we know. Look at what @RoketteS and I said:

>python_x64-6.1_2.7.14-18.spk

**Comment by ymartin59** [comment]
@adispi Can we consider this issue as solved? If so please close.

**Comment by raspberrycoulis** [comment]
How can this be solved when multiple people are reporting that, even though Python 2.7 is installed, packages (including CouchPotato) complain it’s not and won’t start?

Here’s a screenshot of my packages to show that we have installed the correct version of Python:

![Python 2.7 installed](https://user-images.githubusercontent.com/6330169/40266562-89977aa2-5b45-11e8-909d-8622aa62345c.jpeg)

**Comment by adispi** [comment]
It is my belief that it can only be resolved when the package name "python" will be added to the repository....

**Comment by sibyy** [comment]
To voice everyone else here, the issue is that officially Python 2.7 is not available for Apollolake (e.g. DS918+). Most of us got Python installed from thread #3138.

But now the updates to SABnzbd and CouchPotato  are looking for a different/official version of Python I guess? 

As suggested just above, python need to be officially added to the repository for Apollolake.

Thanks

**Comment by ymartin59** [comment]
@adispi OK so may you please detail me a specific failure with a package name and version, DSM architecture or model, and error logs from `/var/packages/xxx/target/var/log/` to have at least a clue why service does not start even when `python` package is present. Please also mention `python` package version. (in fact details requested in issue creation template you decided to discard...)

**Comment by ymartin59** [comment]
New Python 2.7 package version is not ready to be published... there are troubles with ARM architectures. And I guess no reason why beta/test package version may fail to start published application (yet - please provide service startup logs)...

**Comment by adispi** [comment]
Model: DS918+
CPU Intel Celron J3455
DSM Version: DSM 6.1.6-15266 update 1

packackes: Mylar 20180228-4 + 20180519-5, SABnzbd 2.3.3-34, couchpotato 20180513-7, beets 1.4.3-1, headphones 20180228-7 + 20151012-7, lazylibrarian 20180228-4 + 20140611-4

error: 
the following packages are missing in package server. please contact the synology support team for help.
python.
![image](https://user-images.githubusercontent.com/1352350/40276463-8015ef8a-5c13-11e8-8234-38a3318a16da.png)

there is no folder created for the failed install under /var/package, therefore there is no log file
because of that, and the error not being specific, i do not know the python package version.
however, i do believe it to be 2.7.14-18 and higher.

thank you.
adi.

**Comment by RoketteS** [comment]
Model: DS218+
CPU: INTEL Celeron J3355
DSM Version: DSM 6.1.6-15266 Update 1
packages: Mylar 20180228-4 + 20180519-5, couchpotato 20180513-7

Same error as @adispi 

log files: same as @adispi 

Python used is from here: python_x64-6.1_2.7.14-18.spk

**Comment by RoketteS** [comment]
Do you also want the CouchPotato.log file?

**Comment by ymartin59** [comment]
About CouchPotato log, root cause is that service port is already in use by either another instance of CouchPotato or another service:
```
server.listen(config['port'], config['host'])
error: [Errno 98] Address already in use
```
So from my point of view, nothing related to `python` availability or not.
Have you installed `couchpotatoserver-custom` at the same time?

**Comment by raspberrycoulis** [comment]
So, I’ve just tried something here. I have two NAS drives now, the DS215j and the DS718+. I’m using the 718+ for my main tasks and the 215j as a backup NAS. Basically I upgraded from the 215j to the 718+ and found the issue with CP complaining about the missing Python package when migrating my settings. 

However, I’ve just tried installing CP on my 215j and it “failed to install” but when I looked at the logs located at `/var/packages/couchpotatoserver/target/var/logs` I noticed it appeared to be running so I tried accessing it in the browser and it is indeed running. Odd. 

However, I’m still getting the missing Python issue on my 718+ despite manually installing it via the .spk file mentioned several times in here. 

Not sure if this helps or not but I wanted to share in case.

**Comment by RoketteS** [comment]
@ymartin59 as I already thought is had nothing to do with interference from -custom and the standard couchpotato. 
1) Did a clean uninstall and it worked
2) Install coach potato from the .spk and it worked
Still the update is pending after installation and selecting the update will warn that python is not installed.

**Comment by ymartin59** [comment]
I have tested a mixed installation on x64 xpenology: after couchpotato asking for lacking python, I installed it manually from #3138 and then couchpotato accepts to install and run.
My hypothesis is that DSM checks availability from online repository instead of locally installed packages.
I cannot publish a more recent version for only apollolake, it will mask previous version to other architectures.
The only relevant thing to do is to fix ARM support for Python 2.7.14 so that it can be published

**Comment by ymartin59** [comment]
May you please test again to install Python from #3138 and test latest CouchPotato after switching Package Center Settings "Trust Level" to "Any publisher".

**Comment by remirr** [comment]
@ymartin59 I downloaded the latest 6.1 packed, did a manual install on my NAS and so I assumed it updated the current package. Then tried to update the two existing packaged, couchpotato and SAB but still get the same error message saying python is missing.

**Comment by ymartin59** [comment]
I have just published Python 2.7.14 - should fix your install/upgrade troubles

**Comment by Drakuhl** [comment]
Well, it fixed it partially. Python is indeed now visible to the other packages, so I updated Mylar...
...which now fails to start: "Failed to run package service"
It worked fine before,
